src/NUnit-Executor/NUnitExecutor.cs

```diff
@@ -1,61 +1,169 @@
 ﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Xml;
+using Logger;
 using NTestController;
-using System.Collections.Generic;
+using NUnitReader;
+using Utilities;
 
 namespace NUnitExecutor
 {
-    public class NUnitExecutorPlugin : IPlugin
+    public class NUnitExecutorPlugin : IExecutorPlugin
     {
-        private IReadOnlyOptions _options;
-        private IComputer _computer;
-        private TestQueue _testQueue;
-        private string _nunitPath;
+        private static ILogger Log { get; } = ConsoleLogger.Instance;
+
+        private string NunitPath => Computer.NunitPath;
+        private string _xmlConfig;
+
+        #region Constructors
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="NUnitExecutor.NUnitExecutorPlugin"/> class.
+        /// Initializes a new instance of the <see cref="NUnitExecutorPlugin"/> class.
         /// </summary>
-        /// <param name="options">Options from the command line.</param>
-        /// <param name="computer">Target Computer where the tests will be run.</param>
-        /// <param name="testQueue">Test queue.</param>
-        /// <param name="nunitPath">Nunit path.</param>
-        public NUnitExecutorPlugin(
-            IReadOnlyOptions options,
-            IComputer computer,
-            TestQueue testQueue,
-            string nunitPath)
+        /// <param name="xmlConfig">NTestController.xml file path.</param>
+        public NUnitExecutorPlugin(string xmlConfig)
         {
-            _options = options;
-            _computer = computer;
-            _testQueue = testQueue;
-            _nunitPath = nunitPath;
+            ThrowIf.StringIsNullOrWhiteSpace(xmlConfig, nameof(xmlConfig));
+
+            _xmlConfig = xmlConfig;
         }
 
-        #region Inherited from IPlugin
+        #endregion Constructors
+
+        #region Inherited from IExecutorPlugin
+
+        public string Name { get; private set; }
+
+        public PluginType PluginType
+        {
+            get { return PluginType.TestExecutor; }
+        }
 
-        public string Name { get; }
-        public PluginType PluginType { get; }
+        public IReadOnlyOptions Options { get; set; }
+        public IComputer Computer { get; set; }
+        public TestQueue TestQueue { get; set; }
 
         /// <seealso cref="IPlugin.Execute()"/>
         public bool Execute()
         {
-            var test = _testQueue.DequeueTestToRun();
+            Log.WriteInfo("===== Starting tests on Computer: {0}", Computer.Hostname);
+
+            var test = TestQueue.DequeueTestToRun() as NUnitTest;
 
             while (test != null)

```

Comment:
Before this while loop, print a warning if test == null saying no NUnit Tests were found.